src/value.rs

```diff
@@ -33,7 +33,7 @@ pub(super) fn f64_to_bytes(f: f64) -> u128 {
 /// operations and fat pointers. This idea was taken from rustc's trans.
 #[derive(Clone, Copy, Debug)]
 pub enum Value {
-    ByRef(Pointer),
+    ByRef(PrimVal),
```

Comment:
> This is the key change, right? Everything else essentially just makes the types work out again?

Pretty much, yea.

> Things would be a little nicer if this was not necessary, but I suppose with ZSTs we don't really have an alternative.

This should also make implementing the intptrcast easier ;)